app/src/main/java/it/niedermann/owncloud/notes/main/MainActivity.java

```diff
@@ -591,21 +594,26 @@ public void onIconClick(NavigationItem item) {
     }
 
     @Override
-    public void applyBrand(int mainColor, int textColor) {
-        applyBrandToPrimaryToolbar(activityBinding.appBar, activityBinding.searchToolbar);
-        applyBrandToFAB(mainColor, textColor, activityBinding.fabCreate);
-
-        binding.headerView.setBackgroundColor(mainColor);
-        binding.appName.setTextColor(textColor);
-        activityBinding.progressCircular.getIndeterminateDrawable().setColorFilter(getSecondaryForegroundColorDependingOnTheme(this, mainColor), PorterDuff.Mode.SRC_IN);
+    public void applyBrand(int color) {
+        final var util = BrandingUtil.of(color, this);
+        util.material.themeFAB(activityBinding.fabCreate);
+        util.platform.colorCircularProgressBar(activityBinding.progressCircular);
+        util.platform.colorNavigationView(binding.navigationView);
+        util.notes.applyBrandToPrimaryToolbar(activityBinding.appBar, activityBinding.searchToolbar, colorAccent);
+
+        binding.headerView.setBackgroundColor(color);
+        @ColorInt final int headerTextColor = ColorUtil.INSTANCE.getForegroundColorForBackgroundColor(color);
+        binding.appName.setTextColor(headerTextColor);
+        DrawableCompat.setTint(binding.logo.getDrawable(), headerTextColor);
 
         // TODO We assume, that the background of the spinner is always white
-        activityBinding.swiperefreshlayout.setColorSchemeColors(contrastRatioIsSufficient(Color.WHITE, mainColor) ? mainColor : Color.BLACK);
-        binding.appName.setTextColor(textColor);
-        DrawableCompat.setTint(binding.logo.getDrawable(), textColor);
+        activityBinding.swiperefreshlayout.setColorSchemeColors(contrastRatioIsSufficient(Color.WHITE, color) ? color : Color.BLACK);
```

Comment:
There is a `AndroidXViewThemeUtils.themeSwipeRefreshLayout` function available. Was it not working for this specific screen?

Reply:
Just overlooked it :wink:

Reply:
NP. I have some work to do to improve discoverability of stuff in the ui lib.